### DISCUSSION TITLE
LiveChart2 inside TemplatedControl Won't Render

### DISCUSSION THREAD

**Discussion by bsampica** [AvaloniaUI/Avalonia#12528] [Q&A] [answered]
Hey Guys, this looks eerily similar to something I faced years ago, when I was working with early versions of XAML based Silverlight.  

Here's the general setup.

Create a templated control, and setup a LiveChart2 Inside.   (I'm going to clip the code to the respective components to keep it brief)

```
<ControlTemplate>
   <lvc:PieChart Height="300" Width="300" InitialRotation="-225" MaxAngle="270" Total="100" Series="{TemplateBinding ChartSeriesData }">
</ControlTemplate>
```

Then, back in the CodeBehind for the TemplatedControl I use:
```
public IEnumerable<ISeries> ChartSeriesData 
    {
        get => GetValue(ChartSeriesDataProperty);
        set => SetValue(ChartSeriesDataProperty, value, BindingPriority.LocalValue);
    }

    public static readonly StyledProperty<IEnumerable<ISeries>> ChartSeriesDataProperty =
        AvaloniaProperty.Register<GraphDashboardControl, IEnumerable<ISeries>>(nameof(ChartSeriesData));
```

In the MainWindow.xaml I expose the control like this:
```
<cc:GraphDashboardControl
                LabelText="Tank 1" ChartSeriesData="{Binding ChartData}" /> // ChartData from the MainWindowViewModel.xaml =>
```

```
public IEnumerable<ISeries> ChartData =>
        new GaugeBuilder()
            .WithLabelsSize(50)
            .WithInnerRadius(75)
            .WithBackgroundInnerRadius(75)
            .WithBackground(new SolidColorPaint(new SKColor(100, 181, 246, 90)))
            .WithLabelsPosition(PolarLabelsPosition.ChartCenter)
            .AddValue(30, "gauge value", SKColors.YellowGreen, SKColors.Red) // defines the value and the color 
            .BuildSeries();Ï
```

I Also have a label in the custom control (not shown) and setup the back end property for the text the same way.  It renders just fine, but the LiveChart right next to it will never show.  

If feels almost like the data binding is happening to late, or it can't detect the property changed between the IEnumerable() which didn't change, and the Series inside it, which DID change. I don't know how how to trigger the change on the main property later in the binding process.  I've tried setting BindingPriority.Animation, or Template, but nothing really changed anything.

Any advice from anyone?  Thank you in advance.

**Comment by timunie** [comment]
@bsampica can you try with latest nightly? we had a fix recently for PolyLine maybe it solves also this. But not sure. If not, please upload a very minimal sample for us to check.

**Reply by bsampica** [reply]
Thanks @timunie - I'll give it a shot and report back.

**Comment by bsampica** [comment, marked as answer]
> @bsampica can you try with latest nightly? we had a fix recently for PolyLine maybe it solves also this. But not sure. If not, please upload a very minimal sample for us to check.

This looks fixed, thanks @timunie - nice work
